File: saralam_webapp/src/types/job.ts
export interface JobListItem {
  id: string
  title: string
  budget_type: string
  budget_min?: number
  budget_max?: number
  currency: string
  event_date?: string
  status: string
  category_id: string
  seeker_id: string
  created_at: string
  category_name?: string
  applications_count?: number
  new_applications_count?: number
  city?: string
  /** For provider: distance from provider's location (km) */
  distance_km?: number
  /** Premium jobs visible only to paid subscribers */
  visibility?: 'public' | 'premium_only'
  application_deadline?: string
}

export type JobStatus =
  | 'draft'
  | 'open'
  | 'in_review'
  | 'assigned'
  | 'completed'
  | 'cancelled'

export interface JobDetail extends JobListItem {
  description?: string
  requirements?: string[]
  event_date?: string
  duration_hours?: number
  duration_days?: number
  application_deadline?: string
  location?: { address?: string; city?: string; radius_km?: number }
  visibility?: 'public' | 'premium_only'
  slots_available?: number
  media_urls?: string[]
  created_at: string
  updated_at?: string
}

export interface JobCreatePayload {
  category_id: string
  title: string
  description?: string
  requirements?: string[]
  budget_type: 'fixed' | 'per_hour' | 'negotiable'
  budget_min?: number
  budget_max?: number
  currency?: string
  event_date?: string
  duration_hours?: number
  duration_days?: number
  application_deadline?: string
  location?: { address?: string; city?: string; radius_km?: number }
  visibility?: 'public' | 'premium_only'
  slots_available?: number
  media_urls?: string[]
  status?: 'draft' | 'open'
}

File: backend/app/schemas/job.py
from datetime import datetime
from uuid import UUID
from pydantic import BaseModel, Field


class JobListResponse(BaseModel):
    id: UUID
    title: str
    budget_type: str
    budget_min: float | None
    budget_max: float | None
    currency: str
    event_date: datetime | None
    status: str
    category_id: UUID
    seeker_id: UUID
    created_at: datetime

    class Config:
        from_attributes = True


class JobDetailResponse(BaseModel):
    id: UUID
    seeker_id: UUID
    category_id: UUID
    title: str
    description: str | None
    requirements: str | None
    budget_min: float | None
    budget_max: float | None
    budget_type: str
    currency: str
    event_date: datetime | None
    event_duration_hours: float | None
    event_location_id: UUID | None
    status: str
    visibility: str
    preferred_experience_years: str | None
    slots_available: int
    application_deadline: datetime | None
    created_at: datetime
    # seeker_contact when allowed

    class Config:
        from_attributes = True


class JobCreate(BaseModel):
    category_id: UUID
    title: str = Field(..., max_length=500)
    description: str | None = None
    requirements: str | None = None
    budget_min: float | None = None
    budget_max: float | None = None
    budget_type: str = Field(..., pattern="^(fixed|hourly|negotiable)$")
    currency: str = "INR"
    event_date: datetime | None = None
    event_duration_hours: float | None = None
    event_location_id: UUID | None = None
    visibility: str = Field("public", pattern="^(public|subscription_only)$")
    preferred_experience_years: str | None = None
    slots_available: int = 1
    application_deadline: datetime | None = None


class JobUpdate(BaseModel):
    title: str | None = None
    description: str | None = None
    requirements: str | None = None
    budget_min: float | None = None
    budget_max: float | None = None
    budget_type: str | None = None
    event_date: datetime | None = None
    event_duration_hours: float | None = None
    event_location_id: UUID | None = None
    visibility: str | None = None
    preferred_experience_years: str | None = None
    slots_available: int | None = None
    application_deadline: datetime | None = None
    status: str | None = None
